### PR TITLE
Add template for the config files

### PR DESCRIPTION
**Pull request by jamescallumyoung**
Add a template for the lib conf files. These files will define the plugins and extends for the lib. The optionalExtends is there for the future feature where extenda A is only added if extends B is added.

### CODE REVIEW

File: template/conf-template.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "plugins": [
+    "PLUGIN_NAME"
+  ],
+  "extends": [
+    "RULESET_NAME"
+  ],
+  "optionalExtends": [
+    { "if": "RULESET_NAME", "include": "RULESET_NAME" }
+  ]
+}
```
Comment (jamescallumyoung) [inline template/conf-template.json:13]:
```suggestion

}
```

File: template/conf-template.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    "PLUGIN_NAME"
+  ],
+  "extends": [
+    "RULESET_NAME"
+  ],
+  "optionalExtends": [
+    { "if": "RULESET_NAME", "include": "RULESET_NAME" }
+  ]
+
+}
```
Comment (jamescallumyoung) [inline template/conf-template.json:14]:
```suggestion
}

```

**Comment by jamescallumyoung** [comment]
I remove the versions as that behaviour is undefined.

**Review by vabatta — APPROVED** [review]
I think that for the initial use case we have it all good 👍

**Pull request merged**